Apps/W1/APIV2/test/src/APIV2FALocationsE2E.Codeunit.al

```diff
@@ -0,0 +1,161 @@
+        TargetURL := LibraryGraphMgt.CreateTargetURL('', Page::"APIV2 - FA Locations", ServiceNameTxt);
+        LibraryGraphMgt.PostToWebService(TargetURL, FALocationJSON, Response);
+
+        // [THEN] The Fixed Asset Location has been created in the database with all the details.
+        FALocation.Get(TempFALocation.Code);
+        VerifyProperties(Response, FALocation);
+    end;
+
+    [Test]
+    procedure TestModifyFALocation()
+    var
+        FALocation: Record "FA Location";
+        TempFALocation: Record "FA Location" temporary;
+        RequestBody: Text;
+        Response: Text;
+        TargetURL: Text;
+    begin
+        // [SCENARIO] User can modify name in a Fixed Asset Location through a PATCH request.
+        Initialize();
+
+        // [GIVEN] A Fixed Asset Location exists with a name.
+        CreateFALocation(FALocation);
+        TempFALocation.TransferFields(FALocation);
+        FALocation.Code := LibraryUtility.GenerateRandomCodeWithLength(FALocation.FieldNo(Code), Database::"FA Location", 10);
+        TempFALocation.Name := LibraryUtility.GenerateGUID();
+        RequestBody := GetFALocationJSON(TempFALocation);
+
+        // [WHEN] The user makes a patch request to the service and specifies name field.
+        TargetURL := LibraryGraphMgt.CreateTargetURL(FALocation.SystemId, Page::"APIV2 - FA Locations", ServiceNameTxt);
+        LibraryGraphMgt.PatchToWebService(TargetURL, RequestBody, Response);
+
+        // [THEN] The response contains the new values.
+        VerifyProperties(Response, TempFALocation);
+
+        // [THEN] The Fixed Asset Location in the database contains the updated value.
+        FALocation.Get(FALocation.Code);
+        Assert.AreEqual(FALocation.Name, TempFALocation.Name, 'Names should be equal.');
+    end;
+
+    [Test]
+    procedure TestDeleteFALocation()
+    var
+        FALocation: Record "FA Location";
+        FALocationCode: Code[10];
+        TargetURL: Text;
+        Response: Text;
+    begin
+        // [SCENARIO] User can delete a Fixed Asset Location by making a DELETE request.
+        Initialize();
+
+        // [GIVEN] A Fixed Asset Location exists.
+        CreateFALocation(FALocation);
+        FALocationCode := FALocation.Code;
+
+        // [WHEN] The user makes a DELETE request to the endpoint for the Fixed Asset Location.
+        TargetURL := LibraryGraphMgt.CreateTargetURL(FALocation.SystemId, Page::"APIV2 - FA Locations", ServiceNameTxt);
+        LibraryGraphMgt.DeleteFromWebService(TargetURL, '', Response);
+
+        // [THEN] The response is empty.
+        Assert.AreEqual('', Response, 'DELETE response should be empty.');
+
+        // [THEN] The Fixed Asset Location is no longer in the database.
+        FALocation.SetRange(Code, FALocationCode);
+        Assert.IsTrue(FALocation.IsEmpty(), 'Fixed Asset Location should be deleted.');
+    end;
+
+    local procedure CreateFALocation(var FALocation: Record "FA Location")
```

Comment:
Could this procedure be a candidate for a library method? I think it would fit into "Library - Fixed Asset".

Reply:
Again agree but since this is exact the same structure as in Location Tests - I think this should stay as is.

![image](https://github.com/microsoft/ALAppExtensions/assets/106300783/a748146b-69e3-4542-8d28-f5b337cef735)